### DISCUSSION TITLE
"Playbook [...] could not be found" error

### DISCUSSION THREAD

**Discussion by maido** [semaphoreui/semaphore#1280] [Q&A]
Hello everyone, I installed semaphore following the docker compose indications, no issues there, configured project, environment, repository etc.
Repo is configured in a local Gitlab environment, connecting through ssh, connection works fine as I can see the full repo being downloaded inside the docker image, with all my playbooks in its root folder.
So, my task config just specified the repo and the name of the playbook in the field, say "pb_apt-update.yml".
When the job runs, I get a "ERROR! the playbook: pb_apt-update.yml could not be found". I checked inside the docker as I said, repo's there, playbook file is there in its root. What am I doing wrong?
Posting full output of dryrun and verbose run:

``` 9:21:48 AM
Task 17 added to queue
9:21:50 AM
No roles/requirements.yml file found. Skip galaxy install process.
9:21:50 AM
Preparing: 17
9:21:50 AM
Prepare TaskRunner with template: apt_update
9:21:50 AM
Warning: Permanently added '[storage.gitlab.xxx.yy]:10022' (ED25519) to the list of known hosts.
9:21:50 AM
Updating Repository ssh://git@storage.gitlab.xxx.yy:10022/ansible/ansible-root.git
9:21:50 AM
Warning: Permanently added '[storage.gitlab.xxx.yy]:10022' (ED25519) to the list of known hosts.
9:21:50 AM
From ssh://storage.gitlab.xxx.yy:10022/ansible/ansible-root
9:21:50 AM
* branch master -> FETCH_HEAD
9:21:50 AM
Already up to date.
9:21:50 AM
Get current commit hash
9:21:50 AM
Get current commit message
9:21:50 AM
installing static inventory
9:21:50 AM
No collections/requirements.yml file found. Skip galaxy install process.
9:21:55 AM
Started: 17
9:21:55 AM
Run TaskRunner with template: apt_update
9:21:56 AM
ansible-playbook [core 2.13.6]
9:21:56 AM
config file = None
9:21:56 AM
ERROR! the playbook: pb_apt-update.yml could not be found
9:21:56 AM
configured module search path = ['/tmp/semaphore/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
9:21:56 AM
ansible python module location = /usr/lib/python3.10/site-packages/ansible
9:21:56 AM
ansible collection location = /tmp/semaphore/.ansible/collections:/usr/share/ansible/collections
9:21:56 AM
executable location = /usr/bin/ansible-playbook
9:21:56 AM
python version = 3.10.11 (main, Apr 6 2023, 01:17:26) [GCC 11.2.1 20220219]
9:21:56 AM
jinja version = 3.0.3
9:21:56 AM
libyaml = True
9:21:56 AM
No config file found; using defaults
9:21:56 AM
Running playbook failed: exit status 1```

**Comment by jrglynn2** [comment]
In your Task Template configuration, did you use the fully qualified path to the playbook? Such as `/ansible/playbooks/pb_apt-update.yml`?

**Reply by maido** [reply]
Mmm no the playbook name in this instance is just "pb_apt-update.yml", I entered the shell inside the docker container and I see the repo has been succesfully downloaded in `/tmp/semaphore/repository_1_1/` , in which folder I cans see my `pb_apt-update.yml` playbook, so not sure what fully qualified path should I be putting in that field, any indication? I feel like semaphore should already know where to find that file if you configure a repo for it as I did but maybe I am wrong.

**Reply by jrglynn2** [reply]
Right, but the directory is still needed in the task template is what I am saying.

I use a local directory as my repository, so in the Repositories section, my GIT URL is `/opt/ansible`

All my playbooks are under the `/playbooks/` directory.

So in my Task Template, for "Playbook Filename", I have to enter `/opt/ansible/playbooks/playbook.yml`, or else Semaphore can't find the playbook.

Perhaps this doesn't apply to you, but that is what I had to do.

**Reply by maido** [reply]
OH don't get me wrong, I did test what you said after I read it so thanks for that, I tried every variation even putting the temporary folder inside the docker, no dice, still can't find it, not really sure what I am doing wrong.

**Comment by ansibleguy** [comment]
Greetings!

Correct me if I've misread:

* You are using the semaphore repository-type 'git-ssh' and it seems to be cloned correctly
* You have a task-template that refers to this repository and the playbook 'pb_apt-update.yml'

Just to catch any edge-case - can you please supply the full 'task-template' and 'repository' config you use? (text or screenshots)

**Reply by maido** [reply]
Of course, here goes, i blacked out domains:

![image](https://github.com/ansible-semaphore/semaphore/assets/408160/7420550b-a324-4d5f-90b6-04cbf0302e05)

![image](https://github.com/ansible-semaphore/semaphore/assets/408160/14badd23-e68c-4018-a80d-d4d60330d5c6)

The environment contains:

```
{
  "ansible_user": "root",
  "ansible_python_interpreter": "/usr/bin/env python3"
}
```

I hope it's what you needed to know, let me know if I can provide any additional info, thank you.

**Reply by ansibleguy** [reply]
Hmm.
Normally Ansible-Playbook should find the playbook as:
* the current working directory is set to the absolute repository path here: [AnsiblePlaybook.go#L24](https://github.com/ansible-semaphore/semaphore/blob/develop/lib/AnsiblePlaybook.go#L24)
* the Playbook seems to be passed to Ansible 1-to-1 (relative path) [runner.go#L673](https://github.com/ansible-semaphore/semaphore/blob/develop/services/tasks/runner.go#L673)

So for this to go wrong - I would think - some change in the execution environment has to happen.
Can I ask you if you can just try to remove the `ansible_python_interpreter` from the environment? It is currently the only thing that catches my eye
